src/tools/run-make-support/src/lib.rs

```diff
@@ -201,6 +202,33 @@ pub fn set_host_rpath(cmd: &mut Command) {
     });
 }
 
+/// Clear all non-essential env vars. Behaves like [`std::process::Command::env_clear`] but
+/// retains essential env vars:
+///
+/// - `TMP` or `TEMP` for Windows.
+/// - `TMPDIR` for non-Windows platforms.
+///
+/// Note that this will clear all previously-set env vars on the [`Command`].
+pub fn clear_non_essential_env_vars(cmd: &mut Command) {
```

Comment:
In theory, this *could* be a `CommandExt` extension trait, but that seems more complicated to read/maintain, so I opted for a free function instead.